python-backend/app/api/mines/mine/models/mine.py

```diff
@@ -5,6 +5,11 @@
 from ....utils.models_mixins import AuditMixin, Base
 from app.extensions import db
 
+# NOTE: Be careful about relationships defined in the mine model. lazy='joined' will cause the relationship
+# to be joined and loaded immediately, so that data will load even when it may not be needed. Setting
+# lazy='select' will lazy load that data when the property is first accessed. There are other options as well
+# that may be best in different situations: https://docs.sqlalchemy.org/en/latest/orm/loading_relationships.html
+
 
 class Mine(AuditMixin, Base):
     __tablename__ = 'mine'
```

Comment:
Hmm 🤔 My understanding was that `joined` only loaded the relationship but not the data until you access it.
I may be wrong though. Excellent job on exploring and adding more performing options

Reply:
No, lazy='joined' is a form of eager loading that will include those tables as part of a join statement:
select from mine m join mine_locaion ml on m.id=ml.id

lazy='selectin' is a form of eager loading that runs a seperate query to load the data instead of joining

lazy='select' is actual lazy loading

In my experiments, lazy='joined' is faster for 1:1 relationships, and lazy='selectin' is faster for 1:* relationships if you want to eager load.